src/test/scala/org/broadinstitute/dsde/workbench/leonardo/dao/MockGoogleDataprocDAO.scala

```diff
@@ -23,6 +23,10 @@ class MockGoogleDataprocDAO(protected val dataprocConfig: DataprocConfig, protec
 
   private def googleID = UUID.randomUUID()
 
+  def getEmailFromAccessToken(accessToken: String)(implicit executionContext: ExecutionContext): Future[String] = {
+    Future{"user1@example.com"}
```

Comment:
We should make this return a different user under some conditions (e.g. if `accessToken == "unauthorized"`) and add a case in `ProxyRoutesSpec` verifying that a 401 is returned.

There's already a test for _absence of a cookie_: https://github.com/broadinstitute/leonardo/blob/develop/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutesSpec.scala#L89-L94. This would be testing if a cookie is passed for a user not on the whitelist.

Reply:
Also add LeoRoutesSpec test cases for checking the whitelist.